docs/book/getting-started/system-architectures.md

```diff
@@ -0,0 +1,159 @@
+
+In this scenario, all ZenML services are hosted on infrastructure hosted by the ZenML Team.
+Customer secrets and credentials required to access customer infrastructure are
+stored and managed by the ZenML Pro Control Plane.
+
+On the ZenML Pro infrastructure, only ML _metadata_ (e.g. pipeline and
+model tracking and versioning information) is stored. All the actual ML data
+artifacts (e.g. data produced or consumed by pipeline steps, logs and
+visualizations, models) are stored on the customer cloud. This can be set up
+quite easily by configuring
+an [artifact store](../component-guide/artifact-stores/artifact-stores.md)
+with your MLOps stack.
+
+Your tenant only needs permissions to read from this data to display artifacts
+on the ZenML dashboard. The tenant also needs direct access to parts of the
+customer infrastructure services to support dashboard control plane features
+such as CI/CD, triggering and running pipelines, triggering model deployments
+etc.
+
+The advantage of this setup is that it is a fully-managed service, and is
+very easy to get started with. However, for some clients even some metadata
+can be sensitive: These clients should refer to the other architecture diagram.
+
+<details>
+
+<summary>Detailed Architecture Diagram for SaaS deployment</summary>
+
+<figure><img src="../.gitbook/assets/cloud_architecture_saas_detailed.png" alt="ZenML Pro Full SaaS deployment"><figcaption>ZenML Pro Full SaaS deployment with ZenML secret store</figcaption></figure>
+
+</details>
+
+We also offer a hybrid SaaS option where customer secrets are stored on the
+customer side. In this case, the customer connects their own
+secret store directly to the ZenML server that is managed by us. All ZenML
+secrets used by running pipelines to access infrastructure services and
+resources are stored in the customer secret store. This allows users to
+use [service connectors](../how-to/auth-management/service-connectors-guide.md)
+and the [secrets API](../how-to/interact-with-secrets.md) to authenticate
+ZenML pipelines and the ZenML Pro to 3rd party services and infrastructure
+while ensuring that credentials are always stored on the customer side.
+{% endhint %}
+
+![ZenML Pro SaaS deployment with Customer secret store](../.gitbook/assets/cloud_architecture_scenario_1_1.png)
+
+<details>
+
+<summary>Detailed Architecture Diagram for SaaS deployment with custom secret store configuration</summary>
+
+<figure><img src="../.gitbook/assets/cloud_architecture_saas_detailed_2.png" alt="ZenML Pro Full SaaS deployment with custom secret store"><figcaption>ZenML Pro Full SaaS deployment with customer secret store</figcaption></figure>
+
+</details>
+
+### ZenML Pro Self-Hosted Architecture
+
+![ZenML Pro self-hosted deployment](../.gitbook/assets/cloud_architecture_scenario_2.png)
+
+In this scenario, all services, data, and secrets are deployed on the customer
+cloud. This is the opposite of Scenario 1, and is meant for customers who
```

Comment:
We haven't mentioned scenarios in the document yet, so it feels a bit out of place...